tasks/logging.yml

```diff
@@ -0,0 +1,75 @@
+---
+- name: Validate package_state
+  assert:
+    that: package_state in ['present', 'absent']
+    msg: "Received invalid package state: '{{ package_state }}'. The Cloud Ops Ansible role supports the following package states: 'present' and 'absent'."
+
+
+- name: Create temp directory
+  tempfile:
+    path: /tmp
+    state: directory
+    suffix: cloud_ops_shell_scripts
+  register: tempfolder
+  check_mode: no
+  changed_when: false
+
+- name: Download script
+  get_url:
+    url: https://dl.google.com/cloudagents/add-logging-agent-repo.sh
+    dest: "{{ tempfolder.path }}/add-logging-agent-repo.sh"
+    mode: 0755
+  check_mode: no
+  changed_when: false
+
+- when: package_state == 'present'
+  block:
+  - name: Add repo and install agent
+    command:
+      chdir: "{{ tempfolder.path }}"
+      cmd: "bash add-logging-agent-repo.sh --also-install --version={{ version }} {{ '--dry-run' if ansible_check_mode else '' }}"
+    environment:
+      REPO_CODENAME: "{{ ansible_distribution_release }}"
+    register: result
+    check_mode: no
+    changed_when: "'No changes made.' not in result.stdout_lines"
+    notify: restart logging agent
+
+  - name: Copy main config file onto the remote machine
+    copy:
+      src: "{{ main_config_file }}"
+      dest: /etc/google-fluentd/google-fluentd.conf
+      force: yes
+      mode: 0644
+      validate: '/sbin/google-fluentd -tC %s'
+    when: main_config_file|length > 0
```

Comment:
Can you just use main_config_file as a truthy value directly? ("when: main_config_file")

Reply:
Done.